lib/events/dynamoevents/dynamoevents.go

```diff
@@ -499,46 +500,58 @@ func (l *Log) SearchEvents(fromUTC, toUTC time.Time, filter string, limit int) (
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	input := dynamodb.QueryInput{
-		KeyConditionExpression:    aws.String(query),
-		TableName:                 aws.String(l.Tablename),
-		ExpressionAttributeValues: attributeValues,
-		IndexName:                 aws.String(indexTimeSearch),
-	}
-	start := time.Now()
-	out, err := l.svc.Query(&input)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-	g.WithFields(log.Fields{"duration": time.Since(start), "items": len(out.Items)}).Debugf("Query completed.")
+
+	var lastEvaluatedKey map[string]*dynamodb.AttributeValue
 	var total int
-	for _, item := range out.Items {
-		var e event
-		if err := dynamodbattribute.UnmarshalMap(item, &e); err != nil {
-			return nil, trace.BadParameter("failed to unmarshal event for %v", err)
+	for i := 0; i < backend.DefaultLargeLimit/100; i++ {
```

Comment:
define a new constant instead of using a multiple of something unrelated

Reply:
Again, just trying to follow precedent: https://github.com/gravitational/teleport/blob/master/lib/backend/dynamo/dynamodbbk.go#L389

This was also suggested by Sasha in an earlier comment. Personally I agree though, since I don't know the background behind `DefaultLargeLimit` and why it should be divided by 100 in these cases.

Reply:
@pierrebeaucamp I think we agreed to use hardcoded 100mb value for the buffer size (suggested by Sasha) per request instead of `DefaultLargeLimit`. 

```go
for downloaded < maxBufferSize  {

    // 900mb is dynamo db max page size
    downloaded += 900; 
}
```

Reply:
Not all existing code is good code that should be followed.
Let's do what makes the most sense for the problem at hand.

The doc comment on `DefaultLargeLimit` is:
```
	// DefaultLargeLimit is used to specify some very large limit when limit is not specified
	// explicitly to prevent OOM
```
which is not very meaningful and can't possibly be the right value for all types of "limit"